src/main/java/io/quarkus/qlue/Chain.java

```diff
@@ -40,7 +41,8 @@
     Chain(final ChainBuilder chainBuilder) throws ChainBuildException {
         // copy information from chainBuilder so it can be safely reused
         this.classLoader = chainBuilder.classLoader;
-        final Set<StepBuilder> steps = new HashSet<>(chainBuilder.steps);
+        final Set<StepBuilder> steps = Collections.newSetFromMap(new IdentityHashMap<>());
```

Comment:
Would be better to pre-size here:

```suggestion
        final Set<StepBuilder> steps = Collections.newSetFromMap(new IdentityHashMap<>(chainBuilder.steps.size()));
```

Reply:
Ah good catch! Thanks!